Cython/Compiler/Options.py

```diff
@@ -391,6 +392,7 @@ class DEFER_ANALYSIS_OF_ARGUMENTS:
     'total_ordering': ('cclass', ),
     'dataclasses.dataclass' : ('class', 'cclass',),
     'cpp_locals': ('module', 'function', 'cclass'),  # I don't think they make sense in a with_statement
+    'ufunc': ('function',),
```

Comment:
We probably also need to make sure that this directive isn't inherited by nested functions.

Reply:
Good spot. I've fixed it and added a test.